ledger/complete/wal/checkpoint_v6_writer.go

```diff
@@ -0,0 +1,529 @@
+package wal
+
+import (
+	"bufio"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/ledger/complete/mtrie/flattener"
+	"github.com/onflow/flow-go/ledger/complete/mtrie/node"
+	"github.com/onflow/flow-go/ledger/complete/mtrie/trie"
+	utilsio "github.com/onflow/flow-go/utils/io"
+)
+
+const subtrieLevel = 4
+const subtrieCount = 1 << subtrieLevel
+
+type NodeEncoder func(node *trie.MTrie, index uint64, scratch []byte) []byte
+
+type resultStoringSubTrie struct {
+	Index     int
+	Roots     map[*node.Node]uint64 // node index for root nodes
+	NodeCount uint64
+	Checksum  uint32
+	Err       error
+}
+
+// StoreCheckpointV6 stores checkpoint file into a main file and 17 file parts.
```

Comment:
Since it is concurrent now, you can consider adding tracing here so you can visualize store/load in grafana.  Thisi may simplify finding potential bottlenecks.